app/views/reports/regions/index.html.erb

```diff
@@ -22,9 +23,22 @@
         </div>
 
         <script>
+          const search = new RegionsSearch().search;
+          const form = $("#search-form")
+          const input = form.find("#search_input")
+          form.on("submit", function(event) {
+            search(input);
+            return false;
+          })
+          form.on("input", function(event) {
+            search(input);
+            return false;
+          })
```

Comment:
Handle both the submit and the input event, which captures any change in the search field as well as just hitting "enter" on the form itself.